src/main/kotlin/com/pubnub/api/crypto/CryptoModule.kt

```diff
@@ -0,0 +1,93 @@
+package com.pubnub.api.crypto
+
+import com.pubnub.api.PubNubException
+import com.pubnub.api.crypto.cryptor.AesCbcCryptor
+import com.pubnub.api.crypto.cryptor.Cryptor
+import com.pubnub.api.crypto.cryptor.HeaderParser
+import com.pubnub.api.crypto.cryptor.LegacyCryptor
+import com.pubnub.api.crypto.cryptor.ParseResult
+import com.pubnub.api.crypto.data.EncryptedData
+
+class CryptoModule internal constructor(
+    internal val primaryCryptor: Cryptor,
+    internal val cryptorsForEncryptionOnly: List<Cryptor> = emptyList(),
```

Comment:
As discussed having Map<ByteArray, Cryptor> is problematic because of the fact that two ByteArrays having the same content are not consider equal so it make hard to retrieve  element from map by id.